File: testcases/x86_testcases/test3.py
def factorial(x : int) -> int:
    if x <= 1:
        return 1
    else:
        return x * factorial (x - 1)

def main() -> None:
    x : int = 10
    i : int = 0
    for i in range(x + 1):
        y : int = factorial(i)
        print("Value of factorial of ", i, " is equal to ", y)
    return

if __name__ == "__main__":
    main()

File: testcases/x86_testcases/x86_output/test3/test3.c
#include <stdio.h>

int factorial(int x) {
    if(x <= 1) return 1;
    else return (x * factorial(x-1));
}

void main2() {
    int x = 10;
    for(int i=0; i < (x+1); i++) {
        int y = factorial(i);
        printf("Value of factorial of ");
        printf("%d", i);
        printf(" is equal to ");
        printf("%d", y);
        printf("\n");
    }
    return;
}

int main() {
    main2();
    return 0;
}
